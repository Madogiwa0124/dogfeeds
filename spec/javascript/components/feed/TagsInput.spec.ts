import { mount } from "@vue/test-utils";
import Component from "@js/components/feed/TagsInput.vue";
import { buildTag } from "@spec/mocks/Tags";

describe("components/feed/FeedCard.vue", () => {
  describe("初期表示", () => {
    it("snapshot", () => {
      const props = { recommendedTags: [buildTag("tag1"), buildTag("tags2")] };
      const wrapper = mount(Component, { propsData: props });
      expect(wrapper.element).toMatchSnapshot();
    });

    describe("recommendedTagsが空の場合", () => {
      it("snapshot", () => {
        const props = { recommendedTags: [] };
        const wrapper = mount(Component, { propsData: props });
        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
