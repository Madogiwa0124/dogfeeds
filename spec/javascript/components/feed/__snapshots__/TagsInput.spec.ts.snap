// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/feed/FeedCard.vue 初期表示 recommendedTagsが空の場合 snapshot 1`] = `
<div
  class="tags-input"
>
  <label
    class="label"
  >
    タグ
  </label>
  <!--v-if-->
  
  
  <div
    class="tag-list"
  >
    
    
  </div>
  <input
    class="tag-input-area input is-small"
  />
  <p
    class="help"
  >
    RSSフィードのタグを入力してください。(任意)
  </p>
</div>
`;

exports[`components/feed/FeedCard.vue 初期表示 snapshot 1`] = `
<div
  class="tags-input"
>
  <label
    class="label"
  >
    タグ
  </label>
  <div
    class="recommended-tag-list"
  >
    <span
      class="tag is-light control has-text-weight-medium"
    >
      人気のタグ
    </span>
    
    <span
      class="tag clickable is-success"
    >
      tag1
    </span>
    <span
      class="tag clickable is-success"
    >
      tags2
    </span>
    
  </div>
  
  
  <div
    class="tag-list"
  >
    
    
  </div>
  <input
    class="tag-input-area input is-small"
  />
  <p
    class="help"
  >
    RSSフィードのタグを入力してください。(任意)
  </p>
</div>
`;
